### ISSUE TITLE
Timeline upload status

### ISSUE THREAD

**Issue by aarontp** [google/timesketch#322]
There doesn't seem to be an easy way to tell when a given timeline is done being imported other than to continue to refresh the page until the event counts stop increasing or the count matches the count from the original file.  It would be nice to have something on the timelines tab that indicates when it has completed importing.

**Comment by patrickdamery** [comment]
If no one else is working on this I'd like to work on it. Any advice on where to begin?

**Comment by adamjnichols** [comment]
The js upload function [here](https://github.com/google/timesketch/blob/master/timesketch/ui/api/api.service.ts#L320) actually does the post, from which you might be able to obtain bytes sent vs. total bytes, and instrument them into [this](https://github.com/google/timesketch/blob/master/timesketch/ui/core/upload.directive.ts) directive embedded into [this](https://github.com/google/timesketch/blob/master/timesketch/ui/core/upload.html) view.

**Comment by patrickdamery** [comment]
Ok got it. I've got another question though, I followed the installation instructions and installed Timesketch with Docker, everything worked fine and I was able to access the program successfully. However there seems to be a problem loading /static/dist/bundle.css and /static/dist/bundle.js. Neither of these files seem to exist in the codebase, is this a known issue?

**Comment by adamjnichols** [comment]
Yes, addressed in PR #520.  If you like, you can clone the branch ajn-docker-tweaks and you'll be fixed right up.

**Comment by patrickdamery** [comment]
I cloned the PR and after removing my previous Docker image I ran docker-compose up again. It appeared to try and bind to port 80 for some reason which was a problem because I had apache running so I shut it down. After that Docker seemed to be able to run but there are still problems. Glancing through the logs it seems like there is a problem with postgres:
postgres_1       | 2018-01-22 21:57:04.337 UTC [1] FATAL:  database files are incompatible with server
postgres_1       | 2018-01-22 21:57:04.337 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 9.6, which is not compatible with this version 10.1.

Also I found this error:
timesketch_1     | /docker-entrypoint.sh: line 51: exec: timesketch: not found

For some reason it seemed to have created the random password at least 3 different times which is a behavior I don't remember seeing in the master branch. Any idea as to what would cause this?

**Comment by adamjnichols** [comment]
Your postgres issue will stem from the fact that you have a postgres install on your host OS that is incompatible with the version being started in Docker.  All data files in the Docker version are persisted locally to the host via Docker volumes.  See [this](https://github.com/google/timesketch/tree/ajn-docker-tweaks/docker#where-is-my-data-stored) doc.

The continual generation of the password will be due to the failure of the timesketch_1 container to start, and looping on init until you stop it.  There should be another error there, though this may be due to your Docker env.

Docker is pretty complex, and not in a good way.  All "good" states for containers are interleaved to maximize caching and minimize overhead and container portability.  What this means is that /var/lib/docker will often contain artifacts that are undesirable in a volatile or development environment.  For this reason, it's best to do the following (WARNING: this will remove all containers on this host.  If you do not wish to do this, stop now):

```
sudo systemctl stop docker
sudo rm -fR /var/lib/docker
sudo systemctl start docker
cd timesketch/docker
sudo docker-compose up
```

**Comment by patrickdamery** [comment]
I don't have postgres installed on my host OS, I checked to make sure by running `which psql` and got nothing back.

To be honest this is the first time I've used Docker but I've heard good things from it so I wanted to try it out, so the only Docker container I would have had on my host OS is the main branch of timesketch. I've gone ahead and followed your advice.

Everything worked correctly this time so I'm guessing the postgres issue was caused by the last docker container I had. Also Timesketch is not listening on port 5000 but on port 80 which is fine but just thought I'd point it out as the installation instructions say it'll be in port 5000.

With this I can get to work, I'll let you know if I have any other questions.
